e2e/webhook_serve.js

```diff
@@ -27,8 +28,74 @@ server.post('/slack_compatible_message_response', postSlackCompatibleMessageResp
 server.post('/send_message_to_channel', postSendMessageToChannel);
 server.post('/post_outgoing_webhook', postOutgoingWebhook);
 
+server.post('/send_oauth_credentials', postSendOauthCredentials);
+server.get('/start_oauth', getStartOAuth);
+server.get('/complete_oauth', getCompleteOauth);
+server.post('/postOAuthMessage', postOAuthMessage);
+
 server.listen(port, () => console.log(`Webhook test server listening on port ${port}!`)); // eslint-disable-line no-console
 
+let appID;
+let appSecret;
+let client;
+let authedUser;
+function postSendOauthCredentials(req, res) {
+    appID = req.body.appID.trim();
+    appSecret = req.body.appSecret.trim();
+    client = new ClientOAuth2({
+        clientId: appID,
+        clientSecret: appSecret,
+        authorizationUri: getBaseUrl() + '/oauth/authorize',
+        accessTokenUri: getBaseUrl() + '/oauth/access_token',
+        redirectUri: getWebhookBaseUrl() + '/complete_oauth',
+    });
+    return res.status(200).send('OK');
+}
+
+function getStartOAuth(req, res) {
+    return res.redirect(client.code.getUri());
```

Comment:
Looks like my suggestion is not correct. Could you please check again as I'm still getting error while the webhook server is running:
```
TypeError: Cannot read property 'code' of undefined
    at getStartOAuth (/Users/username/go/src/github.com/mattermost/mattermost-webapp/e2e/webhook_serve.js:56:32)
```

Reply:
On my local is working without any problem. The reason for client to be undefined would be that the credentials were never sent (in other words, the previous test failed to execute completely). Can you see if there is any problem there?

Reply:
Can't see any problem. All tests are passing but still throwing said errors.